File: openbullet2-web-client/src/app/main/dtos/job/proxy-check-job-options.dto.ts
import { ProxyCheckTargetDto } from './proxy-check-job.dto';
import { AbsoluteTimeStartConditionDto, RelativeTimeStartConditionDto } from './start-condition.dto';

export interface ProxyCheckJobOptionsDto {
  name: string;
  startCondition: RelativeTimeStartConditionDto | AbsoluteTimeStartConditionDto;
  bots: number;
  groupId: number;
  checkOnlyUntested: boolean;
  target: ProxyCheckTargetDto | null;
  timeoutMilliseconds: number;
  checkOutput: DatabaseProxyCheckOutput;
}

export enum ProxyCheckOutputType {
  Database = 'databaseProxyCheckOutput',
}

export interface DatabaseProxyCheckOutput {
  _polyTypeName: ProxyCheckOutputType.Database;
}

File: OpenBullet2.Web/Dtos/Job/ProxyCheck/ProxyCheckJobOptionsDto.cs
namespace OpenBullet2.Web.Dtos.Job.ProxyCheck;

/// <summary>
/// The options of a proxy check job.
/// </summary>
public class ProxyCheckJobOptionsDto
{
    /// <summary>
    /// The name of the job.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// When the job should start.
    /// </summary>
    public object StartCondition { get; set; } = new RelativeTimeStartConditionDto();

    /// <summary>
    /// The amount of bots that will check the proxies concurrently.
    /// </summary>
    public int Bots { get; set; } = 1;

    /// <summary>
    /// The ID of the proxy group to check.
    /// </summary>
    public int GroupId { get; set; } = -1;

    /// <summary>
    /// Whether to only check the proxies that were never been tested.
    /// </summary>
    public bool CheckOnlyUntested { get; set; } = true;

    /// <summary>
    /// The target site against which proxies should be checked.
    /// </summary>
    public ProxyCheckTargetDto? Target { get; set; } = null;

    /// <summary>
    /// The maximum timeout that a valid proxy should have, in milliseconds.
    /// </summary>
    public int TimeoutMilliseconds { get; set; } = 10000;

    /// <summary>
    /// The options for the output of a proxy check.
    /// </summary>
    public object CheckOutput { get; set; } = new DatabaseProxyCheckOutputOptionsDto();
}
